flytekit/types/flyte_directory.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+
+import os
+import typing
+from pathlib import Path
+
+from flytekit.annotated.context_manager import FlyteContext
+from flytekit.annotated.type_engine import TypeEngine, TypeTransformer
+from flytekit.models import types as _type_models
+from flytekit.models.core import types as _core_types
+from flytekit.models.literals import Blob, BlobMetadata, Literal, Scalar
+from flytekit.models.types import LiteralType
+
+
+class FlyteDirectory(os.PathLike):
+    """
+    Please first read through the comments on the FlyteFile class as the implementation here is similar.
+
+    One thing to note is that the os.PathLike type that comes with Python was used as a stand-in for FlyteFile.
+    That is, if a task returns an os.PathLike, Flyte takes that to mean FlyteFile. There is no easy way to
+    distinguish an os.PathLike where the user means a File and where the user means a Directory. As such, if you
+    want to use a directory, you must declare all types as FlyteDirectory. You'll still be able to return a string
+    literal though instead of a full-fledged FlyteDirectory object assuming the str is a directory.
+
+    Use cases as inputs:
+        def t1(in1: FlyteDirectory):
+            ...
+
+        def t1(in1: FlyteDirectory["svg"]):
+            ...
+
+    As outputs:
+
+    The contents of this local directory will be uploaded to the Flyte store.
+        return FlyteDirectory("/path/to/dir/")
+
+        return FlyteDirectory["svg"]("/path/to/dir/", remote_path="s3://special/output/location")
```

Comment:
what does it mean for a FlyteDirectory to have a type like svg?

Reply:
all files are svg? I deally I think I want to create something called as a flyteDataset, where there is a dataset index file, for every file in the dataset, each with some properties etc

Reply:
it's not really clear, but I wanted to support it cuz the IDL supports it.